src/main/java/io/smartcat/ranger/core/DiscreteWeightedValue.java

```diff
@@ -0,0 +1,83 @@
+package io.smartcat.ranger.core;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.commons.math3.distribution.EnumeratedDistribution;
+import org.apache.commons.math3.util.Pair;
+
+/**
+ * Randomly selects one of the provided values following the provided weights.
+ *
+ * @param <T> Type this value would evaluate to.
+ */
+public class DiscreteWeightedValue<T> extends Value<T> {
```

Comment:
It would be more uniform - this random generator would have the same constructor as others (domain and distribution). And we'd get rid of that helper pair class.